compiler/rustc_passes/src/check_attr.rs

```diff
@@ -2082,6 +2087,112 @@ impl<'tcx> CheckAttrVisitor<'tcx> {
         false
     }
 
+    fn check_rustc_intrinsic_const_vector_arg(
```

Comment:
this seems unnecessary. Just ICEing the compiler when the attribute is misused is fine.

Reply:
Why is it okay to ICE here while other rustc intrinsics have a check function?

Reply:
The other intrinsics don't need these check functions. Some people just really like adding them.^^ But it's a bunch of extra work to write and maintain that code, and not really for any good reason.

Reply:
I think the original reason the platform intrinsics had checks in the first place was to allow exposing `extern "platform-intrinsics"` on stable: https://rust-lang.github.io/rfcs/1199-simd-infrastructure.html That plan didn't end up making it and stdarch was introduced later. And much later portable-simd was introduced which internally used platform-intrinsics.

Reply:
I'm not entirely sure how to implement this comment, is this to remove the checking completely? If so where would the checking go? And what errors should be thrown and where?

Reply:
This check can be largely removed without replacement. The users of the attribute might ICE if the attribute isn't of the right shape, but that's not a big problem.